File: leetcode-submissions/2364_Count_Number_of_Bad_Pairs.ts
function countBadPairs(nums: number[]): number {
    const mp:Map<number,number>=new Map();
    let totalPairs:number =0;
    mp.set(nums[0],1);
    for(let j:number =1;j<nums.length;j++){
        const currentNumsCount :number = mp.get(nums[j]-j) || 0 ;
        const numsBeforeJ:number = j;
        const badPairs:number = numsBeforeJ-currentNumsCount;
        totalPairs+=badPairs;
        mp.set(nums[j]-j, (mp.get(nums[j]-j) || 0)+1);

    }
    return totalPairs;
};

File: leetcode-submissions/2364_Count_Number_of_Bad_Pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
       unordered_map<int,int>mp;
       int totalPairs=0;
       mp[nums[0]]=1;
       for(int j=1;j<nums.size();j++){
        int totalNumsBeforeJ= j;
        int badPairs = totalNumsBeforeJ-mp[nums[j]-j];
        mp[nums[j]-j]++;
        totalPairs+=badPairs;
       }
       return totalPairs;
    }
};